dev/com.ibm.ws.logging_2_fat/fat/src/com/ibm/ws/logging/fat/MessageTraceFileNameTimedRolloverTest.java

```diff
@@ -0,0 +1,273 @@
+
+        // Restore the initial contents of bootstrap.properties
+        if (bootstrapFile != null && initialBootstrapProps != null) {
+            FileOutputStream out = getFileOutputStreamForRemoteFile(bootstrapFile, false);
+            writeProperties(initialBootstrapProps, out);
+        }
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        if (server != null && server.isStarted()) {
+            server.stopServer(EXPECTED_FAILURES);
+        }
+    }
+
+    /**
+     * Test 1: Configure TimedLogRollover (in server.env) and set traceFileName=stdout in bootstrap.properties,
+     * verify if trace.log is not created, only message.log is rolled over.
+     */
+    @Test
+    public void testTraceFileNameStdoutBootstrapWithTimedRollover() throws Exception {
+        // Simple - just start server (bootstrap.properties already has traceFileName=stdout)
+        server.startServer();
+        server.waitForStringInLog("CWWKF0011I");
+
+        boolean traceLogExists = server.fileExistsInLibertyServerRoot("logs/trace.log");
+        assertFalse("REQUIREMENT: trace.log should NOT be created when traceFileName=stdout", traceLogExists);
+
+        boolean messageLogExists = server.fileExistsInLibertyServerRoot("logs/messages.log") ||
+                                   server.fileExistsInLibertyServerRoot("logs/message.log");
+        assertTrue("message log should exist", messageLogExists);
+    }
+
+    /**
+     * Test 2: Configure TimedLogRollover (in server.env) and traceFileName=stdout in server.xml,
+     * verify if the trace.log is NOT rolled over.
+     */
+    @Test
+    public void testTraceFileNameStdoutServerXmlWithTimedRollover() throws Exception {
+        // Use server.xml configuration for this test
+        server.setServerConfigurationFile("server_trace_stdout.xml");
+        server.startServer();
+        server.waitForStringInLog("CWWKF0011I");
+
+        // Check results
+        boolean traceLogExists = server.fileExistsInLibertyServerRoot("logs/trace.log");
+        assertFalse("REQUIREMENT: trace.log should NOT be created when traceFileName=stdout in server.xml", traceLogExists);
+
+        boolean messageLogExists = server.fileExistsInLibertyServerRoot("logs/messages.log") ||
+                                   server.fileExistsInLibertyServerRoot("logs/message.log");
+        assertTrue("message logs should exist for normal operation", messageLogExists);
+    }
+
+    /**
+     * Test 3: Configure TimedLogRollover (in server.env) and set traceFileName=stdout in bootstrap.properties,
+     * and then dynamically update the traceFileName to trace.log in server.xml,
+     * verify if the trace.log gets rolled over.
+     */
+    @Test
+    public void testDynamicTraceFileNameStdoutToTraceLog() throws Exception {
+        // Start with existing bootstrap.properties (already has traceFileName=stdout)
+        server.startServer();
+        server.waitForStringInLog("CWWKF0011I");
```

Comment:
Same comments as above, use isStarted() method to verify and also configure the bootstrap.properties.